helix-core/pom.xml

```diff
@@ -128,30 +128,10 @@ under the License.
       <artifactId>commons-math3</artifactId>
       <version>3.6.1</version>
     </dependency>
-    <dependency>
-      <groupId>commons-codec</groupId>
-      <artifactId>commons-codec</artifactId>
-      <version>1.6</version>
-    </dependency>
-    <dependency>
-      <groupId>com.google.guava</groupId>
-      <artifactId>guava</artifactId>
-      <version>15.0</version>
-    </dependency>
     <dependency>
       <groupId>org.yaml</groupId>
       <artifactId>snakeyaml</artifactId>
-      <version>1.12</version>
-    </dependency>
-    <dependency>
-      <groupId>io.dropwizard.metrics</groupId>
-      <artifactId>metrics-core</artifactId>
-      <version>3.2.3</version>
-    </dependency>
-    <dependency>
-      <groupId>org.apache.httpcomponents</groupId>
-      <artifactId>httpclient</artifactId>
-      <version>4.5.8</version>
+      <version>1.17</version>
```

Comment:
Same comment on yaml. Although with YAML, I think it should be fine to upgrade. It's not as crucial to Helix pipeline as Jackson.

Reply:
Same. It is because "open source libraries aren't always backward compatible and could cause unwarranted code behavior change.". We have different dependency versions for Jackson-core-asl. In run time, this version difference may cause conflict and exceptions like NoSuchClassException. I actually would like to upgrade it to make sure only one version is used in our project to avoid conflicts.